daemon/algod/api/server/v2/simulate.go

```diff
@@ -0,0 +1,205 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package v2
+
+import (
+	"fmt"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/bookkeeping"
+	"github.com/algorand/go-algorand/data/transactions"
+	"github.com/algorand/go-algorand/data/transactions/verify"
+	"github.com/algorand/go-algorand/ledger"
+	"github.com/pkg/errors"
+)
+
+// ==============================
+// > Simulation Ledger
+// ==============================
+
+// LedgerForSimulator is a ledger interface for the simulator.
+type LedgerForSimulator interface {
+	ledger.DebuggerLedgerForEval
+}
```

Comment:
What does wrapping this here buy you?

Reply:
Jason and I discussed including basic "initial state" information (touched account balances, boxes, etc) in addition to a debug trace to make it easier to reason through what happened within the evaluation. If we end up implementing that feature, we would likely require additional ledger methods for accessing blockchain state so I defined this interface in an attempt to be forward-thinking.

Reply:
I don't think that need implies the need for another interface.  The concrete type that you instantiate here can have all sorts of intial state, but since it has the operations that `ledger.DebuggerLedgerForEval` demands, it can be used.  There are no clients making additional demands that need to be codified in another interface.